test/distributed/test_c10d_functional_native.py

```diff
@@ -775,17 +775,18 @@ def setUp(self):
         torch._inductor.config.triton.store_cubin = True
         torch._inductor.config.debug = True
 
-        self.rank = 0
-        self.world_size = 2
         torch.cuda.set_device("cuda:0")
 
-        store = FakeStore()
-        dist.init_process_group(
-            backend="fake",
-            world_size=self.world_size,
-            rank=self.rank,
-            store=store,
-        )
+        if not dist.is_initialized():
+            self.rank = 0
+            self.world_size = 2
```

Comment:
do you also want to add skip for number of GPUs as well since world_size here is 2, we shouldn't run the test unless there are at least 2 GPUs.

Reply:
Er I have no clue, apparently the tests are running just fine on machines that only have 1 GPU, is that expected?

Reply:
oh that is interesting... But I guess it will hang? No?